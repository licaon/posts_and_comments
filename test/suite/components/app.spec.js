import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import App from 'src/components/app';

describe('App component', () => {
  const createComponent = (props) => {
    const wrapper = shallow(
      <App
        {...props}
      />);
    return toJson(wrapper);
  };

  it('should render component correctly', (done) => {
    const component = createComponent();

    expect(component).toMatchSnapshot();
    done();
  });

  it('should render component correctly with props', (done) => {
    const component = createComponent({
      text: 'new text',
    });

    expect(component).toMatchSnapshot();
    done();
  });
});
