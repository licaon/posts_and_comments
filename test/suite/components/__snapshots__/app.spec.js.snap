// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render component correctly 1`] = `
<div>
  Initial structure
</div>
`;

exports[`App component should render component correctly with props 1`] = `
<div>
  new text
</div>
`;
